NeoML/src/TraditionalML/BytePairEncoder.h

```diff
@@ -17,34 +17,37 @@ limitations under the License.
 
 #include <NeoML/NeoMLDefs.h>
 #include <NeoML/TraditionalML/SubwordEncoder.h>
-#include <NeoML/TraditionalML/WordDictionary.h>
 
 namespace NeoML {
 
 // Class that encodes a UTF-8 word using byte-pair-encoding.
 class NEOML_API CBytePairEncoder : public IBytePairEncoder {
 public:
+	// For creating with CreateModel( BytePairEncoderModelName ). Initialize(...) should be used to complete the setup.
 	CBytePairEncoder() = default;
-	// Construction from dictionary of tokens without additional checks.
-	// Generally ctor is used by CBytePairEncoderTrainer.
-	CBytePairEncoder( const CWordDictionary& tokens, bool useEndOfWordToken, bool useStartOfWordToken );
+	// Construction with an empty dictionary. Generally ctor is used by CBytePairEncoderTrainer.
+	CBytePairEncoder( CParams params ) : params( std::move( params ) ) {}
+	// Loads a dictionary without additional checks. Completes the initialization for CBytePairEncoderTrainer
+	void LoadDictionary( const CBPEDictionary& );

```

Comment:
_discussable_ The name of the method is kinda confusing. I think it's better to call it smth like `InitializeUnsafe`. With this name it becomes more clear that this method completes initialization too.